plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5365,6 +5365,47 @@ public void testDeleteRetainsTableHistory()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCreateOrReplaceTable()
```

Comment:
Add a test to BaseConnectorTest, as all connectors should get test coverage for this connector-related feature.

BaseConnectorTest should cover everything that can be covered in a generic manner

- create or replace when table doesn't exist yet
- replacing table data (but schema doesn't change)
- replacing table data with empty set (potential edge case)
- replacing table data and adding new columns, dropping some columns
- replacing table data and arbitrary schema changes to existing columns


To avoid name collision, rename this test case, as it's Iceberg specific (testing snapshots as well0